src/pages/Admin/Venue/DetailsForm.tsx

```diff
@@ -823,8 +867,79 @@ const DetailsFormLeft: React.FC<DetailsFormLeftProps> = ({
     </div>
   );
 
+  const { sovereignVenueId } = useSovereignVenueId({ venueId });
+
+  const { currentVenue: venue } = useConnectCurrentVenueNG(sovereignVenueId);
+
+  const [userStatuses, setUserStatuses] = useState<UserStatus[]>([]);
+
+  const {
+    isShown: hasUserStatuses,
+    show: showUserStatuses,
+    toggle: toggleUserStatus,
+  } = useShowHide(venue?.showUserStatus);
+
+  // Because this is not using the useForm validation. The use effect needs to manually open the dropdown with user statuses.
+  useEffect(() => {
+    if (!venue) return;
+    const venueUserStatuses = venue?.userStatuses ?? [];
+
+    setUserStatuses(venueUserStatuses);
+
+    if (venue.showUserStatus) {
+      showUserStatuses();
+    }
+  }, [showUserStatuses, venue]);
+
+  const removeUserStatus = (index: number) => {
+    const statuses = [...userStatuses];
+    statuses.splice(index, 1);
+    setUserStatuses(statuses);
+  };
+
+  const addUserStatus = () =>
+    setUserStatuses([...userStatuses, { status: "", color: "#4BCC4B" }]);
+
+  const updateStatusColor = (color: string, index: number) => {
+    const statuses = [...userStatuses];
+    statuses[index] = { status: statuses[index].status, color: color };
+    setUserStatuses(statuses);
+  };
+
+  const updateStatusText = (
+    event: React.FormEvent<HTMLInputElement>,
+    index: number
+  ) => {
+    const allUserStatuses = [...USER_STATUSES, ...userStatuses];
+    const statuses = [...userStatuses];
+
+    const userStatusExists = allUserStatuses.find(
+      (userStatus) => userStatus.status === event.currentTarget.value
+    );
+
+    // @debt Move user statuses to useForm with useDynamicInput, add schema for this validation instead
+    if (userStatusExists) {
+      setError("User statuses", "Already exists");
```

Comment:
can we move this error closer to the user statuses component? moreover the message could be more precise, I guess. We could specify what user status is duplicating.

![image](https://user-images.githubusercontent.com/63313732/124403436-73f55e00-dd04-11eb-9f01-065cf1efd33a.png)

Reply:
Looks like even with the new changes this is still not working properly because this is a manual error and the component visualizing these errors is not handling it correctly.